### ISSUE TITLE
Bug: [no-unused-vars] fix results in ts error

### ISSUE THREAD

**Issue by jendrikw** [typescript-eslint/typescript-eslint#11843]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.9.2&fileType=.tsx&code=FAEwpgxgNghgTmABAcygewEYyog3sRRASwDsAXMOAMxgiQAU40BbIgZzAB4AVAPjwKFEVNGgAUASgBciNmTilkAbkEBfYOuBgAHgAc0cMnkSqlQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjxYAEw7oASuhKxUbAsQpQi89EXQAPPJEk0oshRh2RVhk5CkBfEM6A&tokens=false

### Repro Code

```TypeScript
declare global {
  interface Promise<T> {
    foo(): string;
  }
}

export { };
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true,
  }
}
```

### Expected Result

No warnings

### Actual Result

Warning `'T' is defined but never used. 3:21 - 3:22`. But when I remove the `<T>` get an error from typescript: `2428: All declarations of 'Promise' must have identical type parameters. 3:13 - 3:20`. Typescript also won't allow you to rename the type parameters to `_T`.

### Additional Info

See also https://github.com/microsoft/TypeScript/issues/62881 and https://github.com/TanStack/table/discussions/5222

**Comment by bradzacher** [comment]
The rule works single file and always has and always will.
Declaration merging is not considered.

Syntactically the generic is unused and so the rule rightfully reports on it.
Without type information it's impossible to know if your code is actually involved in a declaration merge.

So we report on it always and it's up to you the user to determine if it is a false positive or not. You can use a disable comment to mark the unused variable as required.

**Comment by kirkwaiblinger** [comment]
@jendrikw 

FYI there are quite a few duplicates, see https://github.com/typescript-eslint/typescript-eslint/issues/10399
